kqueen/blueprints/api/generic_views.py

```diff
@@ -181,42 +181,49 @@ def dispatch_request(self, *args, **kwargs):
 class ListView(GenericView):
     methods = ['GET']
     action = 'list'
+
     limit = 0
     offset = 0
+    sort_field = ''
+    sort_order = 'desc'
+
     _objects_total = 0
 
     def _save_objects_range(self, objects):
         self._objects_total = len(objects)
-        sorted_by_date = sorted(objects, key=lambda x: x.created_at, reverse=True)
+        if self.sort_field:
+            objects = self.sort_objects(objects, self.sort_field, self.sort_order)
+
         if self.limit > 0:
-            self.obj = sorted_by_date[self.offset:self.offset + self.limit]
+            self.obj = objects[self.offset:self.offset + self.limit]
         else:
-            self.obj = sorted_by_date
+            self.obj = objects
 
-    def _filter_objects(self, objects, filters):
-        if not filters:
-            return objects
+    def filter_objects(self, objects, filters):
+        return objects  # implement in child class
 
-        def get_name(obj, attr_name):
-            name = getattr(obj, attr_name, '')
-            if attr_name == 'name':
-                return name
-            return getattr(name, 'name', '')
-
-        return list(filter(
-            lambda x: all(filters[key].lower() in get_name(x, key).lower()
-                          for key in filters.keys()),
-            objects
-        ))
+    def sort_objects(self, objects, key, order):
+        return objects  # implement in child class
 
     def set_object(self, *args, **kwargs):
-        supported_filters = ['name', 'provisioner']
+        supported_filters = ['name', 'provisioner', 'engine']
         filters = {}
         for filter_name in supported_filters:
             filter_value = request.args.get(filter_name)
             if filter_value:
                 filters[filter_name] = filter_value
 
+        supported_sort_fields = ['name', 'provisioner', 'engine',
```

Comment:
I suggest to make it a class constant and to convert it to a dictionary (with sorting key as values), so get_sorting_key will not be needed